Project/Source/Components/ComponentParticleSystem.cpp

```diff
@@ -891,8 +1021,20 @@ void ComponentParticleSystem::UpdateGravityDirection(Particle* currentParticle)
 	currentParticle->direction = float3(x, y, z);
 }
 
+void ComponentParticleSystem::UpdateTrail(Particle* currentParticle) {
+	currentParticle->trail->Update(currentParticle->position);
+}
+
+void ComponentParticleSystem::UpdateWitdhTrail() {
+	for (Particle& currentParticle : particles) {
+		currentParticle.trail->DeleteQuads();
+		currentParticle.trail->width = width;
+	}
+}
+
 void ComponentParticleSystem::KillParticle(Particle* currentParticle) {
 	currentParticle->life = -1;
+	RELEASE(currentParticle->trail);
```

Comment:
Move it to Undertaken function, it's where the Particle is deleted. This function is only called by Scripting to put life to -1